internal/provider/resource_consul_cluster.go

```diff
@@ -417,6 +451,22 @@ func setConsulClusterResourceData(d *schema.ResourceData, cluster *consulmodels.
 		return err
 	}
 
+	// self_link is equal to the terraform ID of the resource
+	if err := d.Set("self_link", d.Id()); err != nil {
```

Comment:
Why is the attribute `self_link` needed if it's the same as the terraform ID?

Reply:
The reason it's a separate field is to be explicit to the user that it is a separate property so they aren't confused between, `id` and `cluster_id`. It also protects against the Terraform id format being changed in the future (not likely but 🤷 ). 

That being said, I think it might be best to "re-build" the link using the `newLink` and `linkUrl` functions just so we aren't relying on the `id`.

Reply:
>might be best to "re-build" the link

Makes sense. Will do.

Reply:
Done.